tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/smt/Z3SolverContext.scala

```diff
@@ -767,26 +767,17 @@ class Z3SolverContext(val config: SolverConfig) extends SolverContext with LazyL
       case OperEx(ApalacheInternalOper.distinct, args @ _*) =>
         val (es, ns) = (args.map(toExpr)).unzip
         val distinct = z3context.mkDistinct(es: _*)
-        (distinct.asInstanceOf[ExprSort],
-            ns.foldLeft(1L) {
-              _ + _
-            })
+        (distinct.asInstanceOf[ExprSort], sumExprCount(ns))
```

Comment:
```suggestion
        (distinct.asInstanceOf[ExprSort], ns.sum + 1)
```

Reply:
Oh, I have missed that `Seq.sum` existed. That's definitely a better alternative!